platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/internal/cc/impl/ProblemReportingCrossProjectModelAccess.kt

```diff
@@ -163,19 +162,6 @@ class ProblemReportingCrossProjectModelAccess(
             onIsolationViolation(what)
         }
 
-        override fun equals(other: Any?): Boolean {
-            if (other === this) {
-                return true
-            }
-            if (other == null || other.javaClass != javaClass) {
-                return false
-            }
-            val project = other as ProblemReportingProject
-            return delegate == project.delegate && referrer == project.referrer // do not include `access`
-        }
-
-        override fun hashCode(): Int = delegate.hashCode()
-
         override fun toString(): String = delegate.toString()
```

Comment:
💭 This override seems to make sense for all wrapper types and it should be moved up

Reply:
I remember that somewhere in DynamicObject area it was required to have own implementation on the class, to have correct display name. I'll try to check it out again

Reply:
Moved to the superclass